services/celery/celery.py

```diff
@@ -1,16 +1,129 @@
+        if transfer["status"] == "CONFIRMED":
+            order_manager.update_data(transfer["id"], "withdrawal", "CONFIRMED")
+
+
+@app.task(name="open_order_on_dydx", track_started=True)
+def open_order_on_dydx():
+    global open_position
+    global order_id
+
+    if open_position is False:
+        print(open_position)
+        load_contract = LoadContracts()
+        market_price = load_contract.get_asset_price(
+            "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419"
+        )  # make it dynamic
+        print(market_price)
+        trigger_price = 1750
+        if trigger_price >= market_price:
+            try:
+                dydx_order = DydxOrder()
+                admin = DydxAdmin()
+                position_id = admin.get_position_id()
+                user = admin.get_account()
+                user = vars(user)
+                user_balance = user["data"]["account"]["quoteBalance"]
+                print(user_balance)
+                size = float(user_balance) / market_price
+                print("ve", size)
+                size *= 5
+
+                order_manager = OrderManager()
+
+                order_information = dydx_order.create_order(
+                    {
+                        "position_id": position_id,  # make it dynamic
+                        "market": "ETH-USD",  # make it dynamic
+                        "side": "SELL",
+                        "order_type": "MARKET",
+                        "post_only": "false",
+                        "size": str(round(size, 3)),  # make it dynamic
+                        "price": "1600",  # make it dynac
+                        "limit_fee": "0.4",
+                        "expiration_epoch_seconds": time.time() + SEVEN_DAYS_S + 60,
+                        "time_in_force": "IOC",
+                    }
+                )
+                dydx_order_details = vars(order_information)
+                dydx_order_details = dydx_order_details["data"]["order"]
+                order_manager.store_data(dydx_order_details, "dydx_orders")
+                order_id = dydx_order_details["id"]
+                print("order data-->", dydx_order_details)
+                open_position = True
+            except Exception as e:
+                print("this is error", vars(e))
+    else:
+        print("order is already placed")
+
+
+@app.task(name="close_order_on_dydx", default_retry_delay=4 * 60)
+def close_order_on_dydx():
+    global open_position
+    global order_id
+
+    if open_position is True:
+        print(open_position)
+        load_contract = LoadContracts()
+        market_price = load_contract.get_asset_price(
+            "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419"
+        )  # make it dynamic
+        print(market_price)
+        trigger_price = 1600
+        if trigger_price < market_price:
+            try:
+                dydx_order = DydxOrder()
+                admin = DydxAdmin()
+                position_id = admin.get_position_id()
```

Comment:
position_id doesn't change but we're calling the position id API every time this task runs. Seems too much latency. Can you compute the time taken for this task to run ? 
just add a start_time = time.time() at the start and print(start_time - (time.time() ) )

Reply:
on mainnet ^